### ISSUE TITLE
Adding new areas and freeing them takes forever.

### ISSUE THREAD

**Issue by viksl** [godotengine/godot#67311]
### Godot version

(mono) 4.0 b2

### System information

Windows 10, Nvidia GTX 1660Ti, Vulkan

### Issue description

This is probably not a bug report per se and I know that godot 4.0 is in early beta, I just wanted to bring the subject up and ask about it too, I understand that we are at the phase of bug fixing and optimizations. :-)

This is probably not a bug report but I was testing a 3.5 report about some crash on areas and wanted to see how it works in 4.0, it doesn't crash but the entire project freezes for long when you try to add lots of areas, I've tried two ways:

1. adding 1000 areas in one frame
2. adding 10 areas per frame, total at the end was again 1000

To clarify, all areas are at the same position (overlapping), default settings, added through a script.
Both ways the project freezes for seconds.

I then tried to `QueueFree` all the areas and that also takes a lot of time.

In comparison, doing the same in godot 3.5 it takes fraction of a second or so to add the areas in one frame, freeing them crashes it though.

### Steps to reproduce

Add a number of areas at the same position (higher numbers, for example 1000+).

### Minimal reproduction project

I can add one if needed but is it needed?

**Comment by towai** [comment]
i don't think this necessarily has to do with physics, as i believe i've just encountered the same issue, only with Controls. Comparing with 3.4 was quite eye-opening — i was able to free over 100,000 nodes in around half a second without doing anything complicated (just starting from the back of the parent's `get_children()` array). i haven't tested that on 3.5, though.

With 4.0, though, all things the same, the _speed_ isn't actually all that different, i think. Plus, the freezing isn't really the end of the world for a GUI application as long as it just occurs when loading really big files, but the real showstopper is that on a release build, and _only_ on a release build, it will crash the engine altogether.

In the case of 4.0 release builds it seems like you can free as many nodes as you want in one frame, but then when it comes time to process the next frame, or at least at some stage after the process step where you did a lot of freeing, that's when the hang occurs. somehow, seemingly, even when using `await(get_tree().process_frame)` to spread out the frees, the hang will still occur after you reach some threshold (for me around 1280 or so). i might do more testing later but it's way late.
